src/address_helpers.c

```diff
@@ -0,0 +1,66 @@
+#include <stdint.h>
+#include <stddef.h>
+
+#include "bech32.h"
+#include "globals.h"
+#include "get_private_key.h"
+#include "os.h"
+#include "ux.h"
+
+/* return false in case of error, true otherwise */
+bool get_public_key(uint32_t account_number, uint32_t index, uint8_t *public_key_array) {
+    cx_ecfp_private_key_t private_key;
+    cx_ecfp_public_key_t public_key;
+    bool error = false;
+
+    if (!get_private_key(account_number, index, &private_key)) {
+        return false;
+    }
+
+    BEGIN_TRY {
+        TRY {
+            cx_ecfp_generate_pair(CX_CURVE_Ed25519, &public_key, &private_key, 1);
+        }
+        CATCH_ALL {
+            error = true;
+        }
+        FINALLY {
+            memset(&private_key, 0, sizeof(private_key));
```

Comment:
Sensitive data should be cleared with a function that will never be optimize by the compiler. I suggest using `explicit_bzero` instead of `memset` in that case:

```c
explicit_bzero(&private_key, sizeof(private_key));
```

This change should also be made in the `get_private_key`, `get_public_key`, `sign_message`, `sign_auth_token` and `sign_tx_hash` functions.

Reply:
done, thanks for your suggestion